Makefile

```diff
@@ -1102,16 +1190,34 @@ $(OBJECT_DIR)/$(TARGET)/%.o: %.S
 	$(V1) echo "%% $(notdir $<)" "$(STDOUT)"
 	$(V1) $(CROSS_CC) -c -o $@ $(ASFLAGS) $<
 
-## official            : Build all official (travis) targets
-official: $(OFFICIAL_TARGETS)
 
 ## all               : Build all valid targets
 all: $(VALID_TARGETS)
 
+## official          : Build all official (travis) targets
+official: $(OFFICIAL_TARGETS)
+
+## targets-group-1   : build some targets
+targets-group-1: $(GROUP_1_TARGETS)
+
+## targets-group-2   : build some targets
+targets-group-2: $(GROUP_2_TARGETS)
+
+## targets-group-3   : build some targets
+targets-group-3: $(GROUP_3_TARGETS)
+
+## targets-group-4   : build some targets
+targets-group-4: $(GROUP_4_TARGETS)
+
+## targets-group-test-and-rest :
+##                     build the rest of the targets (not listed in group 1, 2 or 3)
+targets-group-test-and-rest: test $(GROUP_TEST_AND_REST_TARGETS)
```

Comment:
We should not mix two different things anyhow. When this job is in red, it is not obvious if the tests or some target build failed. Unless digging into the log. Split the two different types of jobs. If tests are short and quick, that slot will be free to take new jobs anyhow. Plus the extra overhead.
Structure and order above detailed hand optimization.